File: C/Listas/Lista 01/ex07.c
//Leia 4 números e mostre a média ponderada, sabendo-se que os pesos são respectivamente: 1, 2, 3 e 4.
#include <stdio.h>

int main(void){
    int n1,n2,n3,n4;
    float mediapond, soma;
    printf("Digite o primeiro número: ");
    scanf("%d", &n1);
    printf("Digite o segundo número: ");
    scanf("%d", &n2);
    printf("Digite o terceiro número: ");
    scanf("%d", &n3);
    printf("Digite o quarto número: ");
    scanf("%d", &n4);

    soma= 1+2+3+4;
    mediapond = n1*1+n2*2+n3*3+n4*4;

    printf("A média ponderada entre %d,%d,%d e %d = %.2f",n1,n2,n3,n4,mediapond/soma);
}

File: Java/extra/Lista 01/src/ex07.java
/* Leia 4 números e mostre a média ponderada, sabendo-se que os pesos são 
respectivamente: 1, 2, 3 e 4. */

import java.util.Scanner;

public class ex07 {
    static float media_ponderada(float a) {
        float x = a / (1 + 2 + 3 + 4);
        return x;
    }

    public static void main(String[] args) {
        float num = 0, soma = 0;
        try (Scanner leitor = new Scanner(System.in)) {
            for (int i = 1; i < 5; i++) {
                System.out.format("Digite o %d° número: ", i);
                num = leitor.nextFloat();
                soma+=(i*num);
            }
        }
        System.out.format("\nA média ponderada será de %.2f ", media_ponderada(soma));
    }
}
